core/src/utils.jl

```diff
@@ -327,3 +327,39 @@ function basin_bottoms(
     bottom_b = something(bottom_b, bottom_a)
     return bottom_a, bottom_b
 end
+
+"""
+Replace the truth states in the logic mapping which contain wildcards with
+all possible explicit truth states.
+"""
+function expand_logic_mapping!(logic_mapping::Dict{Tuple{Int, String}, String})::Nothing
+    keys_old = collect(keys(logic_mapping))
+
+    for (node_id, truth_state) in keys_old
+        pattern = r"^[ $(TF*)* ]+$"
+        msg = "Truth state \'$truth_state\' contains illegal characters or is empty."
+        @assert occursin(pattern, truth_state) msg
+
+        n_wildcards = count(==('*'), truth_state)
+
+        if n_wildcards > 0
+            control_state = logic_mapping[(node_id, truth_state)]
+            delete!(logic_mapping, (node_id, truth_state))
+            for substitution in Iterators.product(fill(['T', 'F'], n_wildcards)...)
```

Comment:
I'm struggling to understand how this part works, could use a comment.

Since strings are immutable, they are not very convenient for these type of operations. You could also use a `Vector{UInt8}` after checking that the truth_state is valid:

```
julia> UInt8['T', 'F', '*']
3-element Vector{UInt8}:
 0x54
 0x46
 0x2a

julia> String(ans)  # note: consumes the vector
"TF*"
```